File: MyControlResidence.Api/Entidades/Transacao.cs
using MyControlResidence.Api.Enums;
using System.ComponentModel;

namespace MyControlResidence.Api.Entidades
{
    public class Transacao
    {
        public long Id { get; set; }

        public string Descricao { get; set; }

        public decimal Valor { get; set; }

        public TipoTransacao Tipo { get; set; }

        public long CategoriaId { get; set; }

        public long PessoaId { get; set; }
    }
}

File: src/types/transacao.ts
export interface CreateTransacao{
  descricao:string
  valor:string | number
  tipo:number
  categoriaId:number
  pessoaId:number
}

export interface ResponseTransacao{
  id:number
  descricao:string
  valor:number
  tipo:number
  categoriaId:number
  pessoaId:number
  dataHoraCriacao:string
}